core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/EnvelopeSimContext.kt

```diff
@@ -28,19 +29,42 @@ constructor(
          * *after* the end of the path (or right at the end).
          */
         val dangerPointOffsets: List<Offset<TravelledPath>>,
+        /**
+         * List of ETCS-block detector offsets for every block on the path. Starts at the start of
+         * the path, can end after the end of the path.
+         */
+        val detectorOffsets: List<Offset<TravelledPath>>
     ) {
         /**
-         * Returns the SVL location: next buffer stop or switch, whichever is closest. If there is
-         * any.
+         * Returns the next danger point location: next buffer stop or switch, whichever is closest.
+         * If there is any.
+         */
+        fun getDangerPoint(offset: Offset<TravelledPath>): Offset<TravelledPath>? {
```

Comment:
Nit: is it meant to be eventually used outside the class? Otherwise it could be private